src/main/java/Duke/TaskList/TaskList.java

```diff
@@ -0,0 +1,150 @@
+     */
+    public void deleteTask(int taskIndex) throws DukeException {
+        checkTaskIndex(taskIndex);
+        System.out.println("Noted. I've removed this task:");
+        System.out.println(taskStore.get(taskIndex).toString());
+        taskStore.remove(taskIndex);
+        System.out.println(String.format("Now you have %d tasks in the list", this.getSizeOfTaskList()));
+    }
+
+
+    /**
+     * Checks if position of task given is valid.
+     *
+     * @param taskIndex Position of task on the TaskList to check.
+     * @throws DukeException If no task at taskIndex.
+     */
+    public void checkTaskIndex(int taskIndex) throws DukeException {
+        if (taskIndex >= this.getSizeOfTaskList() || taskIndex < 0) {
+            throw new DukeException("No task at this number, please try again");
+        }
+    }
+
+
+    /**
+     * Return size of A ja/rrayList within the TaskList.
+     *
+     * @return Integer representation of size of ArrayList inside TaskList.
+     */
+    public int getSizeOfTaskList() {
+        return taskStore.size();
+    }
+
+    /**
+     * Prints out all task of the ArrayList if not empty,
+     *
+     * @throws DukeException If existing ArrayList in TaskList is empty.
+     */
+    public void printOutTaskList() throws DukeException {
+        if (taskStore.size() == 0) {
+            throw new EmptyListException("List is empty, please add task");
+        }
+        for (int i = 0; i < taskStore.size(); i++) {
+            System.out.println((i + 1) + "." + taskStore.get(i).toString());
+        }
+    }
+
+    /**
+     * Adds new task into ArrayList inside TaskList.
+     *
+     * @param newTask Task that will be added to the ArrayList of tasks.
+     */
+    public void addTask(Task newTask) {
+        this.taskStore.add(newTask);
+        System.out.println("Got it. I've added this task:");
+        System.out.println("  " + newTask.toString());
+        System.out.println(String.format("Now you have %d tasks in the list", this.getSizeOfTaskList()));
+    }
+
+    /**
+     * Prints list of tasks which contains matching words.
+     *
+     * @param searchWord Word to find within task.
+     * @throws DukeException If searchWord is empty or tasks with matching words not found.
+     */
+    public void findTask(String searchWord) throws DukeException {
+        ArrayList<Task> searchResult = new ArrayList<>();
+        if (searchWord.trim().length() == 0) {
+            throw new DukeException("Please input word to find");
+        }
+
+        for (int i = 0; i < taskStore.size(); i++) {
+            if (taskStore.get(i).toString().contains(searchWord)) {
+                searchResult.add(taskStore.get(i));
+            }
+        }
+        if (searchResult.size() == 0) {
+            throw new EmptyListException("No results found!");
+        }
+        for (int i = 0; i < searchResult.size(); i++) {
+            System.out.println((i + 1) + "." + searchResult.get(i).toString());
+        }
```

Comment:
This is quite similar to `printOutTaskList`. Maybe we can create a `TaskList` here after filtering, then call the method as to not repeat the code.

Reply:
Thank you for this suggestion! Makes the code a lot more efficient and clean👍  8475c55b322b6e5013a713391e12e80d7f5378e9